app/views/declaration_information.scala.html

```diff
@@ -43,11 +43,10 @@
     }
 
     @if(status == SubmissionStatus.ADDITIONAL_DOCUMENTS_REQUIRED && sfusConfig.isSfusEnabled) {
-        @link(text = messages("submissions.sfus"), call = Call("GET", sfusConfig.sfusLink), target="_blank")
+        @link(text = messages("submissions.sfus"), call = Call("GET", s"${sfusConfig.sfusLink}/${submission.mrn.getOrElse("")}"), target="_blank")
```

Comment:
IMO too much Scala logic in this template. You could pass directly the entire url as an `Option[String]` parameter from the controller, where you could in addition check if DMSDOC and isSfusEnabled.
The idea is that in general you should always try to include less/none (wherever feasible of course) Scala logic in a template, so to make easier for a frontend dev who maybe does not know Scala to work with the template.

Reply:
This page is quite dynamic in nature. I could do as you suggest but to do so just for this simple SFUS link would make the approach inconsistent with how the rest of the page is coded. It would also IMO hide the logic of when an SFUS link should be present and when it shouldn't be from the reader (and I don't like that personally). I understand the principle your referring to but our UI guys are familiar with twirl and anyway they have their own prototype apps that they code so I don't see the value of doing this for us in this specific context.